src/mpyl/utilities/s3/__init__.py

```diff
@@ -0,0 +1,113 @@
+"""A client to connect to an S3 bucket"""
+
+import pathlib
+from dataclasses import dataclass
+from typing import Optional
+from logging import Logger
+from os import walk
+from http import HTTPStatus
+import boto3
+from botocore.exceptions import ClientError
+
+from ...project import Project
+from ...steps.models import RunProperties
+
+
+@dataclass
+class S3ClientConfig:
+    service_name = 's3'
+    region_name = 'eu-central-1'
+    access_key_id: Optional[str]
+    secret_access_key: Optional[str]
+    bucket_name: str
+    bucket_root_path: str
+
+    def __init__(self, run_properties: RunProperties, project: Project):
+        s3_connection_info = run_properties.config.get('s3', None)
+        self.access_key_id = s3_connection_info.get('accessKeyId')
+        self.secret_access_key = s3_connection_info.get('secretAccessKey')
+        self.bucket_name = project.s3_bucket.bucket.get_value(run_properties.target)
```

Comment:
might be a bit much to have the entire `Project` as input param while you only need the bucket

Reply:
true - I've updated it to take the bucket name as a parameter